generic_tracers/generic_COBALT.F90

```diff
@@ -496,456 +508,507 @@ subroutine user_add_params
+    !call get_param(param_file, "generic_COBALT", "o2_2_nitrif",   cobalt%o2_2_nitrif,  "o2_2_nitrif",  units="mol O2 mol N-1", default=  &
     !       (106.0*35.0+449.0*8.0)/(16.0+106.0*35.0))                                   ! mol O2 mol N-1
-    call g_tracer_add_param('o2_2_nitrif', cobalt%o2_2_nitrif,2.0)
-    call g_tracer_add_param('o2_2_no3', cobalt%o2_2_no3, 150.0 / 16.0)                       ! mol O2 mol N-1
+    call get_param(param_file, "generic_COBALT", "o2_2_nitrif",   cobalt%o2_2_nitrif,  "o2_2_nitrif",  units=" ", default=2.0)
+    call get_param(param_file, "generic_COBALT", "o2_2_no3",      cobalt%o2_2_no3,     "o2_2_no3",     units="mol O2 mol N-1", default=  150.0 / 16.0)                       ! mol O2 mol N-1
     !
     !-----------------------------------------------------------------------
     ! Nutrient Limitation Parameters (phytoplankton)
     !-----------------------------------------------------------------------
     !
-    call g_tracer_add_param('k_fed_Di', phyto(DIAZO)%k_fed,  4.0e-9)                  ! mol Fed kg-1
-    call g_tracer_add_param('k_fed_Lg', phyto(LARGE)%k_fed,  2.0e-9)                  ! mol Fed kg-1
-    call g_tracer_add_param('k_fed_Md', phyto(MEDIUM)%k_fed, 8.0e-10)                 ! mol Fed kg-1
-    call g_tracer_add_param('k_fed_Sm', phyto(SMALL)%k_fed,  4.0e-10)                 ! mol Fed kg-1
-    call g_tracer_add_param('k_nh4_Lg', phyto(LARGE)%k_nh4,  5.0e-8)                  ! mol NH4 kg-1
-    call g_tracer_add_param('k_nh4_Md', phyto(MEDIUM)%k_nh4, 2.0e-8)                  ! mol NH4 kg-1
-    call g_tracer_add_param('k_nh4_Sm', phyto(SMALL)%k_nh4,  1.0e-8)                  ! mol NH4 kg-1
-    call g_tracer_add_param('k_nh4_Di', phyto(DIAZO)%k_nh4,  1.0e-7)                  ! mol NH4 kg-1
-    call g_tracer_add_param('k_no3_Lg', phyto(LARGE)%k_no3,  2.5e-6)                  ! mol NO3 kg-1
-    call g_tracer_add_param('k_no3_Md', phyto(MEDIUM)%k_no3, 1.0e-6)                 ! mol NO3 kg-1
-    call g_tracer_add_param('k_no3_Sm', phyto(SMALL)%k_no3,  5.0e-7)                  ! mol NO3 kg-1
-    call g_tracer_add_param('k_no3_Di', phyto(DIAZO)%k_no3,  5.0e-6)                  ! mol NO3 kg-1
-    call g_tracer_add_param('k_po4_Di', phyto(DIAZO)%k_po4,  1.0e-7)                  ! mol PO4 kg-1
-    call g_tracer_add_param('k_po4_Lg', phyto(LARGE)%k_po4,  5.0e-8)                  ! mol PO4 kg-1
-    call g_tracer_add_param('k_po4_Md', phyto(MEDIUM)%k_po4, 2.0e-8)                  ! mol PO4 kg-1
-    call g_tracer_add_param('k_po4_Sm', phyto(SMALL)%k_po4,  1.0e-8)                  ! mol PO4 kg-1
-    call g_tracer_add_param('k_sio4_Lg',phyto(LARGE)%k_sio4, 2.0e-6)                  ! mol SiO4 kg-1
-    call g_tracer_add_param('k_sio4_Md',phyto(MEDIUM)%k_sio4,1.0e-6)                  ! mol SiO4 kg-1
-    call g_tracer_add_param('k_fe_2_n_Di', phyto(DIAZO)%k_fe_2_n, 12.0e-6 * 106.0 / 16.0)   ! mol Fe mol N-1
```

Comment:
@charliestock @amoebaliz @andrew-c-ross  
Are these parameter defaults that depend on stoichiometry or is there another reason for the division?

Reply:
Yes - values in the literature are generally given in units of iron to carbon.  Since nitrogen serves as the main currency in COBALT and other BGC models, we covert to iron to nitrogen by using the Redfield ratio of 106 carbons per unit of nitrogen.

Reply:
added `c2n` conversion.